source/taskmanager/src/main/java/nl/aerius/taskmanager/QueueWatchDog.java

```diff
@@ -43,23 +42,21 @@
   public boolean isItDead(final boolean runningWorkers, final int numberOfMessages) {
     boolean doReset = false;
     if (runningWorkers && numberOfMessages == 0) {
-      if (firstProblem == 0) {
-        firstProblem = new Date().getTime();
+      if (firstProblem == null) {
+        firstProblem = now().plusMinutes(RESET_TIME_MINUTES);
       } else {
-        doReset = calculatedDiffTime(new Date().getTime() - firstProblem) > RESET_TIME_MINUTES;
+        doReset = now().isAfter(firstProblem);
       }
     } else {
-      firstProblem = 0;
+      firstProblem = null;
     }
     return doReset;
   }
 
   /**
    * Convinces method for testing, because the time unit is in minutes, testing would take forever.
```

Comment:
Do we want to update this? I am not exactly clear on what it says. Might just be a typo though.

Reply:
I've updated the comment. Hope it's clear now.